contents/why_julia.md

```diff
@@ -69,32 +74,271 @@ Let's proceed then!
+Suppose you want to define two new `struct`s based on a animals:
+
+```jl
+sc(
+"""
+struct fox end
+struct chicken end
+"""
+)
+```
+
+I want to define addition for both `fox` and `chicken` new types.
+We proceed by defining a new function signature of the `+` from the `Base` module of Julia^[this is just an example for teaching purposes. Doing something similar as this example will probably result in a [method invalidation](https://julialang.org/blog/2020/08/invalidations/).]:
+
+```jl
+sco(
+"""
+import Base: +
++(F::fox, C::chicken) = "trouble"
+"""
+)
+```
+
+Now let us call addition with the `+` sign in an instantiated `fox` and `chicken` objects:
+
+```jl
+sco(
+"""
+my_fox = fox()
+my_chicken = chicken()
+my_fox + my_chicken
+"""
+)
+```
+
+This is the power of multiple dispatch: **we don't need everything from scratch for our custom-defined user types**.
+If you are excited as much as we are by multiple dispatch, here are two more in-depth examples.
+The first is a [fast and elegant implementation of a one-hot vector](https://storopoli.io/Bayesian-Julia/pages/1_why_Julia/#example_one-hot_vector) by @storopoli2021bayesianjulia.
+The second is an interview of [Christopher Rackauckas](https://www.chrisrackauckas.com/) at [Tanmay Bakshi YouTube's Channel](https://youtu.be/moyPIhvw4Nk?t=2107) (see from time 35:07 onwards) [@tanmaybakshiBakingKnowledgeMachine2021].
+Chris mentions that, while using [`DifferentialEquations.jl`](https://diffeq.sciml.ai/dev/), a package that he developed and currently maintains, a user reported that Julia has the best GPU-based quaternion ODE Solver and he said that he had no idea because he didn't even implemented it.
+Most of the merit is due to multiple dispatch and high user code/type sharing.
+
+To conclude, we think that multiple dispatch is best explained by one of the creators of Julia, [Stefan Karpinski, at JuliaCon 2019](https://youtu.be/kc9HwsxE1OY) [@thejuliaprogramminglanguageJuliaCon2019Unreasonable2019] (if you reading the printed book or a static PDF please click on the link to go the video or check the citation):
+
+<iframe width="698" height="393" src="https://www.youtube.com/embed/kc9HwsxE1OY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+## Julia in the Wild {#sec:julia_wild}
+
+In @sec:julia_accomplish, we exposed why we think Julia is such a unique programming language.
+We showed simple examples about the main features of Julia.
+If you would like to have a deep dive on how Julia is being used, we have some **interesting use cases**:
+
+1. NASA uses Julia in a supercomputer to analyze the ["Largest Batch of Earth-Sized Planets Ever Found"](https://exoplanets.nasa.gov/news/1669/seven-rocky-trappist-1-planets-may-be-made-of-similar-stuff/) and achieve a whopping **1,000x speedup** to catalog 188 million astronomical objects in 15 minutes.
```

Comment:
Have you also seen <https://discourse.julialang.org/t/julia-and-the-satellite-amazonia-1/57541>? Maybe a fun one too

I'm starting to get why you could convince your uni, lots of things happening in Brazil with Julia!

Reply:
Nice! I had no idea! Added in 0f01951